www/js/app.js

```diff
@@ -1650,6 +1680,31 @@ function archiveReadyCallback (archive) {
             params.originalContentInjectionMode = null;
         }
     }
+    // Set contentInjectionMode to serviceWorker when opening a new archive in case the user switched to Safe Mode/jquery Mode when opening the previous archive
+    if (params.contentInjectionMode === 'jquery') {
+        params.contentInjectionMode = settingsStore.getItem('contentInjectionMode');
```

Comment:
Be careful, the value in settingsStore can be `serviceworkerlocal`. You have to test for that, and if it is `serviceworkerlocal`, you need to set `params.serviceWorkerLocal = true;` and params.contentInjectionMode = `serviceworker`.